paddle/operators/sequence_erase_op.cu

```diff
@@ -41,26 +41,45 @@ __global__ void LabelErasedIdx(const T* in_dat, const int in_len,
   }
 }
 
-template <typename T>
-__global__ void GetOutLod(const T* num_erased, const int* in_lod,
-                          const int lod_len, int* out_lod0) {
+__global__ void GetOutLod(const size_t* num_erased, const size_t* in_lod,
+                          const size_t lod_len, size_t* out_lod0) {
   int index = blockIdx.x * blockDim.x + threadIdx.x;
   if (index < lod_len) {
     out_lod0[index] = in_lod[index] - num_erased[in_lod[index]];
   }
 }
 
 template <typename T>
-__global__ void SetOutput(const T* in_dat, const int in_len,
-                          const int* num_erased, T* out_dat) {
+__global__ void SetOutput(const T* in_dat, const int64_t in_len,
+                          const size_t* num_erased, T* out_dat) {
   int index = blockIdx.x * blockDim.x + threadIdx.x;
   if (index < in_len) {
-    if (in_dat[index] != in_dat[index + 1]) {
+    if (num_erased[index] == num_erased[index + 1]) {
       out_dat[index - num_erased[index]] = in_dat[index];
     }
   }
 }
 
+template <typename T, typename Vector>
+thrust::device_vector<T> set_device_vector(Vector& vector) {
+  thrust::host_vector<T> host_vec(vector.size());
+  for (size_t i = 0; i < vector.size(); ++i) {
+    host_vec[i] = vector[i];
+  }
+  thrust::device_vector<T> dev_vec = host_vec;
+  return dev_vec;
+}
+
+template <typename T>
+std::vector<T> get_std_vector(thrust::device_vector<T>& dev_vec) {
```

Comment:
Please ensure that `Vector` in LoD must be `thrust::host_vector` in `.cu` file. Is it necessary converting device_vector to std::vector?

Reply:
Done